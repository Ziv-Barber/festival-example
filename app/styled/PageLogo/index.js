// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Material-ui stuff:
import MuiPaper from '@material-ui/core/Paper'
import MuiTypography from '@material-ui/core/Typography'

// Our types:
import type { PageLogoPropsType } from './index.types'

/**
 * PageLogo react component.
 * @reactProps {string} className Optional class name for the root element.
 * @reactProps {string} text Parameters for the component.
 * @return New element.
 */
const PageLogo = (props: PageLogoPropsType) => (
  <MuiPaper
    className={props.className || 'pageLogo'}
    elevation={0}
    square={true}
  >
    <MuiTypography className="pageLogoText" variant="h5" component="h2">
      {props.children}
    </MuiTypography>
  </MuiPaper>
)

export default styled(PageLogo)`
  margin: 0 !important;
  min-height: 60vh;
  background-size: cover;
  background-position: center bottom;
  background-image: url(${(props) => props.back});
  display: flex;
  justify-content: center;
  align-items: center;

  & .pageLogoText {
    padding: ${(props) => props.theme.spacing.unit * 4}px !important;
    background: ${(props) => props.theme.palette.secondary.main};
    border: 1px solid #fff;
  }
`

// In case that someone need our types:
export type { PageLogoPropsType as Props }
