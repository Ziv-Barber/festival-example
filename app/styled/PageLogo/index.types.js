// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars

export type PageLogoPropsType = {
  /** Optional class name to use for the root element. */
  className?: string,
  /** Url to the background image. */
  back: string,
  /** The content to display in the middle. */
  children?: React.Node
}
