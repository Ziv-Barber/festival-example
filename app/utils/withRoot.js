// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
