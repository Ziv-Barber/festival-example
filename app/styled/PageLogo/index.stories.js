import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import PageLogo from './index'

storiesOf('PageLogo', module).add(
  'basic',
  () => <PageLogo text="Basic example" />,
  { notes: { markdown: 'This is an example to the PageLogo component.' } }
)
