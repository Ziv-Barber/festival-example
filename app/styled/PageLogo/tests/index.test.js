// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import PageLogo from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {PageLogo} */
describe('<PageLogo />', () => {
  afterEach(cleanup)

  /** @test {PageLogo} */
  it('should match snapshort', () => {
    const CompToTest = () => (
      <PageLogo back="http://example.com/back.jpg">This is a text</PageLogo>
    )
    const App = withRoot(CompToTest)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  /** @test {PageLogo} */
  it('should render', () => {
    const CompToTest = () => (
      <PageLogo back="http://example.com/back.jpg">This is a text</PageLogo>
    )
    const App = withRoot(CompToTest)
    const { getByText } = render(<App />)
    expect(getByText(/^This is a text/)).not.toBeNull()
  })
})
