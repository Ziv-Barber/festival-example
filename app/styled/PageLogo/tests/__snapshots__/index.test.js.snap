// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLogo /> should match snapshort 1`] = `
<div>
  <div
    class="MuiPaper-root-1 MuiPaper-elevation0-3 PageLogo-sc-14ncvse-0 epBoLa"
  >
    <h2
      class="MuiTypography-root-28 MuiTypography-h5-44 pageLogoText"
    >
      This is a text
    </h2>
  </div>
</div>
`;
